service/controller/resource/endpoints/desired.go

```diff
@@ -109,23 +111,38 @@ func (r Resource) searchMasterInstance(ctx context.Context, cr infrastructurev1a
 			},
 		}
 
-		o, err := cc.Client.TenantCluster.AWS.EC2.DescribeInstances(i)
+		o, err := cc.Client.TenantCluster.AWS.EC2.DescribeInstances(instancesInput)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
 
 		if len(o.Reservations) == 0 {
 			return nil, microerror.Maskf(notFoundError, "master instance")
 		}
-		if len(o.Reservations) != 1 {
-			return nil, microerror.Maskf(executionFailedError, "expected one master instance, got %d", len(o.Reservations))
+
+		// check for health of the instance
+		instanceHealthInput := &elb.DescribeInstanceHealthInput{
+			LoadBalancerName: aws.String(key.ELBNameAPI(&cr)),
 		}
-		if len(o.Reservations[0].Instances) != 1 {
-			return nil, microerror.Maskf(executionFailedError, "expected one master instance, got %d", len(o.Reservations[0].Instances))
+
+		o2, err := cc.Client.TenantCluster.AWS.ELB.DescribeInstanceHealth(instanceHealthInput)
+		if err != nil {
+			return nil, microerror.Mask(err)
 		}
 
-		instance = o.Reservations[0].Instances[0]
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d instances and %d elb instnace states", len(o.Reservations), len(o2.InstanceStates)))
```

Comment:
I am not sure what "elb instance states" should tell me when I read the logs. 

```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d instances and %d elb instance states", len(o.Reservations), len(o2.InstanceStates)))
```

Reply:
Oh, this more a really extra debug line for my testing, this can go away.

Reply:
removed